### PR TITLE
GH-678: Data Samplers

### PR DESCRIPTION
**Pull request by alanakbik**
This PR adds a set of Samplers that can be used in the ModelTrainer: 

The `ImbalancedClassificationDatasetSampler` will upsample rare classes and downsample common classes in a classification dataset. It may potentially help with imbalanced datasets. 

The two chunk samplers split a dataset into chunks and shuffle them. This preserves some ordering of the original data while also randomizing the data.

### CODE REVIEW

File: flair/samplers.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+        self.block_size = 1
+        self.plus_window = 0
+        self.epoch_count = 0
+
+    def __iter__(self):
+
+        self.epoch_count += 1
+
+        data = [i for i in range(len(self.data_source))]
```
Comment (bluesheeptoken) [inline flair/samplers.py]: Same as above

**Comment by alanakbik** [comment]
@bluesheeptoken thanks! I've made the changes.

**Comment by alanakbik** [comment]
:+1:

**Comment by yosipk** [comment]
👍

**Pull request merged**